.github/workflows/ci.yml

```diff
@@ -11,45 +11,89 @@ permissions:
 
 jobs:
   check-cs:
-    name: Check Coding Standards
+    name: Coding Standards
     runs-on: ubuntu-latest
     steps:
       - name: Checkout
         uses: actions/checkout@v3
 
-      - name: PHP-CS-Fixer
-        uses: docker://oskarstark/php-cs-fixer-ga
+      - name: Setup PHP
+        uses: shivammathur/setup-php@v2
         with:
-          args: --config=.php-cs-fixer.php --diff --dry-run
+          # Keep in sync with PHP version pinned in tools/php-cs-fixer/composer.json
+          php-version: '8.3'
+
+      - name: Install PHP CS Fixer dependencies
+        uses: ramsey/composer-install@v2
+        with:
+          working-directory: tools/php-cs-fixer
+
+      - name: Run PHP CS Fixer
+        run: tools/php-cs-fixer/vendor/bin/php-cs-fixer fix --config=.php-cs-fixer.php
 
   phpstan:
-    name: PHPStan
+    name: Static Analysis
     runs-on: ubuntu-latest
     steps:
       - name: Checkout
         uses: actions/checkout@v3
 
-      - name: PHPStan
-        uses: docker://oskarstark/phpstan-ga
-        env:
-          REQUIRE_DEV: true
+      - name: Setup PHP
+        uses: shivammathur/setup-php@v2
+        with:
+          php-version: '8.3'
+
+      - name: Install PHPStan dependencies
+        uses: ramsey/composer-install@v2
+        with:
+          working-directory: tools/phpstan
+
+      - name: Install Castor dependencies
+        uses: ramsey/composer-install@v2
+        with:
+          composer-options: --prefer-dist --no-progress --optimize-autoloader --classmap-authoritative
+
+      - name: Run PHPStan
+        run: tools/phpstan/vendor/bin/phpstan --configuration=phpstan.neon
 
   phpunit:
-    name: PHPUnit
+    name: PHPUnit on ${{ matrix.php }} deps ${{ matrix.dependency-versions }}
     runs-on: ubuntu-latest
+    strategy:
+      fail-fast: false
+      matrix:
+        include:
+          - php: "8.1"
+            dependency-versions: "lowest"
```

Comment:
I just realized something. Castor is not a lib but a project. Deps are locked. z don't need to test with lowest composer versions. could you remove this check? Thanks